From 6569de90aa25d5b3bb5be54123976f1235fcf177 Mon Sep 17 00:00:00 2001
From: Steve Kondik <steve@cyngn.com>
Date: Mon, 12 Oct 2015 12:49:32 -1000
Subject: [PATCH 5/5] tinycompress: Fix compilation on old targets

 * Exclude code for codec-specific data if kernel support is missing.

Change-Id: I088e0244b01b01f21b80d9478407a185c34afbe8

# Conflicts:
#	compress.c
---
 compress.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/compress.c b/compress.c
index 1ab2afd..6698096 100644
--- a/compress.c
+++ b/compress.c
@@ -573,6 +573,7 @@ int compress_set_gapless_metadata(struct compress *compress,
 }
 
 #ifdef USE_VENDOR_EXTN
+#if defined(SNDRV_COMPRESS_SET_NEXT_TRACK_PARAM)
 int compress_set_next_track_param(struct compress *compress,
 	union snd_codec_options *codec_options)
 {
@@ -583,6 +584,13 @@ int compress_set_next_track_param(struct compress *compress,
 		return oops(compress, errno, "cannot set next track params\n");
 	return 0;
 }
+#else
+int compress_set_next_track_param(struct compress *compress __unused,
+	union snd_codec_options *codec_options __unused)
+{
+	return 0;
+}
+#endif
 #endif
 
 bool is_codec_supported(unsigned int card, unsigned int device,
-- 
2.17.1

